railties/lib/rails/generators/database.rb

```diff
@@ -90,6 +90,14 @@ def mysql_socket
             "/opt/lampp/var/mysql/mysql.sock"         # xampp for linux
           ].find { |f| File.exist?(f) } unless Gem.win_platform?
         end
+
+        def mysql_database_host
+          if options[:devcontainer]
+            "<%= ENV.fetch(\"DB_HOST\") { \"localhost\" } %>"
+          else
+            "localhost"
+          end
+        end
```

Comment:
Why not using it for all database configs?

```ruby
def database_host
 "<%= ENV.fetch(\"DB_HOST\") { \"localhost\" } %>"
end
```

Would work for `templates/config/databases/postgresql.yml.tt` as well as for all other templates.

But, I am trying to address this issue differently in https://github.com/rails/rails/pull/50975. Have a look.

Reply:
Unfortunately it will not work - for local postgresql on MacOs the connection is by socket using the macos user to authenticate - so no host option is supplied (look at how the `postgres.yml.tt` looks on main to see what I mean). We want need to preserve that behaviour in case the dev is not using a dev container, which is why you see the `if ENV["DB_HOST"]` in the postgres template wrapping the host, user and password fields.